frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -182,14 +182,15 @@ export const parseRuntimeInfoFromRunDetails = (
 
 export const parseRuntimeInfoFromExecutions = (
   taskId: string,
+  taskName: string,
   executions?: Execution[] | null,
 ): PipelineTaskRunStatus | undefined => {
   if (!executions) {
     return undefined;
   }
 
   const execution = executions.find(
-    (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === taskId,
+    (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === (taskName || taskId),
```

Comment:
If both of `taskName` & `taskId` props are required and not possibly `undefined` (since the type is simply `string`, this will always use the taskName. Maybe taskName is meant to be optional and just used to override the usage of taskId?

Reply:
not necessarily. I used `||` which will used the taskId if taskName is `""`, which is what i noticed in cases where the label is empty for some reason.

in JS:
```js
a = "";
b = "defined"
a || b == "defined"
```